server/message_hooks.go

```diff
@@ -640,8 +642,10 @@ func (p *Plugin) Send(teamID, channelID string, user *model.User, post *model.Po
 	}
 
 	timeElapsed := float64((newMessage.CreateAt.UnixMilli() - post.CreateAt)) / float64(1000)
-	p.metricsService.ObserveMessageSyncDuration(actionSourceMattermost, timeElapsed)
-	p.metricsService.ObserveMessagesCount(actionCreated, actionSourceMattermost, isNotDirectMessage)
+	if p.metricsService != nil {
+		p.metricsService.ObserveMessageSyncDuration(actionSourceMattermost, timeElapsed)
+		p.metricsService.ObserveMessagesCount(actionCreated, actionSourceMattermost, isNotDirectMessage)
```

Comment:
I think we changed the name `isNotDirectMessage`. Is this PR not synced with other PRs?

Reply:
yes, we can update it when all the PRs before it gets merged